src/execution_plan/execution_plan.c

```diff
@@ -944,23 +948,28 @@ void ExecutionPlanInit(ExecutionPlan *plan) {
 }
 
 ResultSet *ExecutionPlan_Execute(ExecutionPlan *plan) {
-	Record r;
-	char *err = NULL;
-
+	Record r = NULL;
 	ExecutionPlanInit(plan);
 
-	// Replace the jump context used in ExecutionPlan construction with one for actual execution.
-	jmp_buf *env = QueryCtx_GetEnv();
-	int res = setjmp(*env);
-	if(res != 0) {
-		// Jumped
-		char *err = QueryCtx_GetError();
-		ResultSet_ReportError(plan->result_set, err);
+	/* Replace the exception-handling breakpoint set in ExecutionPlan construction with
+	 * a new breakpoint to capture run-time errors. */
+	jmp_buf *env = QueryCtx_GetExceptionHandler();
+	QueryCtx_SetExceptionHandler(env);
```

Comment:
Why the reset, isn't this call to `QueryCtx_SetExceptionHandler` is unnecessary?

Reply:
:+1: Good catch!